tensorflow_privacy/privacy/membership_inference_attack/seq2seq_mia.py

```diff
@@ -160,28 +193,67 @@ def _get_ranks_for_sequence(logits: np.ndarray,
   return sequence_ranks
 
 
+def _update_batch_loss(batch_logits: np.ndarray,
+                       batch_labels: np.ndarray,
+                       loss: SparseCategoricalCrossentropy):
+  """Updates the loss metric per batch.
+
+  Args:
+    batch_logits: Logits returned by a seq2seq model, dim = (num_sequences,
+    num_tokens, vocab_size).
+    batch_labels: Target labels for the seq2seq model, dim = (num_sequences,
+    num_tokens, 1).
+    loss: SparseCategoricalCrossentropy loss metric.
+  """
+  for sequence_logits, sequence_labels in zip(batch_logits, batch_labels):
+    loss.update_state(sequence_labels.astype(np.float32),
```

Comment:
This is currently modifying one of the input parameters. While this looks good, it can easily lead to other unintended side effects. Can we refactor this to return the new updated loss (like in Pandas or numpy, they always return the result and you need to reassign it). Sam for update_batch_accuracy

Reply:
Sure, I will update both the methods